添加视频点赞点踩、收藏、播放量统计、视频审核、评论审核、弹幕审核、用户历史评论查看、用户历史弹幕查看功能

genius-child-alice opened the pull request:
## 标题
[Feature] 添加视频点赞点踩功能、收藏功能、播放量统计、用户历史评论查看、用户历史弹幕查看、视频审核功能、评论审核功能、弹幕审核功能，涉及对数据库结构的多处添加

genius-child-alice committed:
feat(db): 添加视频点赞和点踩的MongoDB schema

添加VideoUpvoteSchema和VideoDownvoteSchema用于存储视频的点赞和点踩数据
包含必要的字段和唯一索引以防止重复操作

genius-child-alice committed:
将原有的视频点赞模型从 VideoUpvoteSchema.ts 迁移到 VideoVoteSchema.ts
新增视频点踩模型 VideoDownvoteSchemaFactory

genius-child-alice committed:
feat(视频投票): 实现视频点赞和点踩功能及相关服务

添加视频点赞和点踩的核心功能，包括：
- 用户点赞/点踩视频
- 取消点赞/点踩
- 检查用户投票状态
- 自动处理互斥操作（点赞自动取消点踩，反之亦然）

genius-child-alice committed:
feat(视频投票): 添加视频点赞和点踩相关控制器及DTO

实现视频点赞和点踩的控制器功能，包括点赞、取消点赞、点踩和取消点踩四个接口
添加对应的请求和响应DTO类型定义
包含RBAC权限验证和参数处理逻辑

genius-child-alice committed:
feat(路由): 添加视频点赞和点踩相关路由接口

新增四个路由接口用于处理视频点赞和点踩功能：
- POST /video/upvote 用户为视频点赞
- POST /video/downvote 用户为视频点踩
- DELETE /video/upvote/cancel 用户取消点赞
- DELETE /video/downvote/cancel 用户取消点踩

genius-child-alice committed:
feat(视频): 添加视频点赞点踩统计及用户状态功能

在 VideoControllerDto 中添加点赞点踩相关字段
实现获取视频点赞点踩数量的服务方法
公开检查用户点赞点踩状态的方法
在获取视频详情时返回点赞点踩数据及用户状态

genius-child-alice committed:
feat(视频统计): 添加视频播放记录和播放量统计功能

实现视频播放量的精确统计，确保同一用户同一天观看同一视频只计一次播放量
新增 VideoWatchRecordSchema 用于记录观看记录
添加 recordVideoWatchAndIncrementCount 服务处理播放量统计逻辑
在获取视频详情时自动记录播放量

genius-child-alice committed:
feat(收藏夹): 实现收藏夹内容管理功能

添加收藏夹内容管理相关接口，包括添加/移除内容、获取内容列表、更新收藏夹信息、删除收藏夹和调整内容排序
在FavoritesDetailSchema中添加sortOrder字段和组合唯一索引

genius-child-alice committed:
feat(评论弹幕): 添加获取用户个人评论和弹幕功能

新增获取用户个人发布的视频评论和弹幕的接口，包含分页查询功能
在评论和弹幕的数据库模型中添加软删除和审核状态字段
修改现有评论和弹幕查询逻辑，过滤已删除和待审核内容

genius-child-alice committed:
feat(路由): 添加用户屏蔽功能相关路由接口

添加屏蔽用户、关键词、标签、正则表达式及其解封相关的路由接口，并包含获取黑名单列表功能

YasaitoriOvO committed:
Revert "feat(路由): 添加用户屏蔽功能相关路由接口"

This reverts commit e262cf3fdb40bfae1badb06acaec61a8fdb3467e.

genius-child-alice committed:
refactor(FavoritesService): 优化收藏夹内容排序逻辑

重构收藏夹内容排序服务，将原有简单批量更新改为四步处理流程：
1. 读取现有数据并验证
2. 在内存中处理目标排序
3. 计算最终严格1..N顺序
4. 统一落库更新
避免并发操作导致的排序溢出问题，提高排序稳定性

genius-child-alice committed:
feat(收藏夹服务): 添加收藏夹和内容数量上限检查

在创建收藏夹和添加内容到收藏夹时，分别检查用户已有收藏夹数量（上限100个）和收藏夹内内容数量（上限5000个），超过限制时返回错误信息

genius-child-alice committed:
feat(视频服务): 添加视频被收藏数统计功能

在视频详情响应中新增 favoritesCount 字段，统计视频被不同用户收藏的总数（去重计算）
实现 getVideoFavoritesCount 服务方法，使用 MongoDB 聚合管道按用户ID分组统计

genius-child-alice committed:
feat(审核): 新增视频、评论和弹幕的审核功能接口

添加审核相关控制器、服务和DTO，包括：
1. 获取待审核视频/评论/弹幕列表
2. 通过/退回视频/评论/弹幕审核
3. 新增ReviewControllerDto定义审核相关数据结构
4. 实现RBAC权限校验，需要developer角色
5. 添加MongoDB查询和更新逻辑

genius-child-alice committed:
refactor(favorites): 将用户标识从uid改为uuid并添加参数校验

修改收藏夹相关接口和服务的用户标识参数，从使用uid改为使用uuid
为所有控制器添加uuid和token的必填校验
在服务层添加通过uuid获取uid的逻辑

genius-child-alice committed:
fix(ReviewService): 修复类型检查问题，添加类型断言

在评论和弹幕的审核服务中，为查询条件添加类型断言以解决类型检查问题

genius-child-alice committed:
refactor(rbac): 将审核相关功能的权限从 developer 改为 admin 角色

修改审核控制器和服务中的权限检查逻辑，将所有需要 developer 角色的地方改为 admin 角色

genius-child-alice committed:
refactor(ReviewService): 将 admin 角色检查改为 administrator

将代码中所有关于 admin 角色的检查统一改为 administrator，包括函数注释、角色检查逻辑和错误日志信息，保持一致性

cfdxkk commented:
请留意 #79 中对 console.log 的替换，大概率会与 `alice` 分支冲突。

genius-child-alice committed:
refactor(ReviewService): 移除管理员角色检查逻辑

简化审核服务逻辑，移除重复的管理员角色检查代码

genius-child-alice commented:
关于审核功能，我把拥有权限者设置为了administrator角色，然后已经实现仅通过 RBAC 对每个接口的访问控制达成审核权限控制，不再需要 checkUserHasXXXXXXRole 这样的 '静态' 校验函数。
@cfdxkk

genius-child-alice commented:
等#79 中日志记录工具换好了我把几个分支调整一下就行了
@cfdxkk

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
 // https://localhost:9999/video/pending/approved
 // cookie: uid, token
 
+// ========== 新的审核接口（需要 developer 角色，使用 uuid 和 token） ==========
```
```suggestion
// ========== 新的审核接口（使用 uuid 和 token） ==========
```

cfdxkk replied:
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
 
+// ========== 新的审核接口（需要 developer 角色，使用 uuid 和 token） ==========
+
+router.get('/review/video/pending', getPendingReviewVideoListController) // 获取待审核视频列表（需要 developer 角色）
```
与原来的视频审核接口重复了。
```
router.get('/video/pending', getPendingReviewVideoController)
```
你写的这个看起来好一些，旧的可以删掉。但是 Cerasus 和 Lycoris 需要更新。

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
 
+// ========== 新的审核接口（需要 developer 角色，使用 uuid 和 token） ==========
+
+router.get('/review/video/pending', getPendingReviewVideoListController) // 获取待审核视频列表（需要 developer 角色）
```
```suggestion
router.get('/review/video/pending', getPendingReviewVideoListController) // 获取待审核视频列表
```

cfdxkk replied:
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// 	"videoId": 123
+// }
+
+router.post('/review/video/reject', rejectVideoReviewController) // 退回视频审核（需要 developer 角色）
```
KIRAKIRA 是先发后审，退回视频审核目前并不会对视频有任何影响。
打开了议题 #90 供讨论

旧的审核业务设计上未提供“退回审核”的接口，而是直接删除。
``` typescript
router.delete('/video/delete', deleteVideoByKvidController) // 根据视频 ID 删除视频 // WARN: 仅限管理员
```

cfdxkk replied:
```suggestion
router.post('/review/video/reject', rejectVideoReviewController) // 退回视频审核
```
通过 RBAC 限制访问，不用标明接口权限了
至少先把这个注释文本删了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// https://localhost:9999/review/video/pending?num=20&offset=0
+// cookie: uuid, token
+
+router.get('/review/comment/pending', getPendingReviewCommentListController) // 获取待审核评论列表（需要 developer 角色）
```
```suggestion
router.get('/review/comment/pending', getPendingReviewCommentListController) // 获取待审核评论列表
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// https://localhost:9999/review/comment/pending?num=20&offset=0
+// cookie: uuid, token
+
+router.get('/review/danmaku/pending', getPendingReviewDanmakuListController) // 获取待审核弹幕列表（需要 developer 角色）
```
```suggestion
router.get('/review/danmaku/pending', getPendingReviewDanmakuListController) // 获取待审核弹幕列表
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// https://localhost:9999/review/danmaku/pending?num=20&offset=0
+// cookie: uuid, token
+
+router.post('/review/video/approve', approveVideoReviewController) // 通过视频审核（需要 developer 角色）
```
```suggestion
router.post('/review/video/approve', approveVideoReviewController) // 通过视频审核
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// 	"videoId": 123
+// }
+
+router.post('/review/comment/approve', approveCommentReviewController) // 通过评论审核（需要 developer 角色）
```
```suggestion
router.post('/review/comment/approve', approveCommentReviewController) // 通过评论审核
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// 	"commentRoute": "1.2.3"
+// }
+
+router.post('/review/comment/reject', rejectCommentReviewController) // 退回评论审核（需要 developer 角色）
```
```suggestion
router.post('/review/comment/reject', rejectCommentReviewController) // 退回评论审核
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk replied:
现在评论审核业务是通过从 `video-comments` 集合“剪切”数据到 `removed-video-comments` 集合实现，而不是通过修改 `pendingReview`，`userDeletedFlag` 和 `adminDeletedFlag` 标识。

新旧两种实现有些冲突。

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// 	"commentRoute": "1.2.3"
+// }
+
+router.post('/review/danmaku/approve', approveDanmakuReviewController) // 通过弹幕审核（需要 developer 角色）
```
```suggestion
router.post('/review/danmaku/approve', approveDanmakuReviewController) // 通过弹幕审核
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/route/router.ts:
```diff
@@ -475,6 +477,62 @@ router.post('/video/pending/approved', approvePendingReviewVideoController) // 
+// 	"danmakuId": "507f1f77bcf86cd799439011"
+// }
+
+router.post('/review/danmaku/reject', rejectDanmakuReviewController) // 退回弹幕审核（需要 developer 角色）
```
```suggestion
router.post('/review/danmaku/reject', rejectDanmakuReviewController) // 退回弹幕审核
```
通过 RBAC 限制访问，不用标明接口权限了

cfdxkk commented on src/controller/ReviewController.ts:
```diff
@@ -0,0 +1,320 @@
+	const token = ctx.cookies.get('token')
+
+	if (!uuid || !token) {
+		ctx.body = { success: false, message: '参数不合法：缺少 uuid 或 token' }
```
`isPassRbacCheck` 自带参数校验并能够丰富响应体，不需要提前做。
其他地方也需要修改

genius-child-alice committed:
feat: 实现集中式日志服务并替换所有console日志

refactor: 标准化i18n语言标识符处理逻辑

feat: 添加日期工具函数isToday

feat: 添加MongoDB日志schema定义

style: 统一错误处理使用新日志服务

genius-child-alice committed:
feat(审核): 新增审核日志功能及查看接口

实现审核日志记录功能，包括：
1. 创建审核日志MongoDB Schema
2. 在视频/评论/弹幕审核通过/退回时记录日志
3. 新增获取本人审核记录的接口
4. 完善审核相关DTO定义

YasaitoriOvO committed:
🔒️ 修复播放量不正常递增

YasaitoriOvO committed:
🐛 修复导入相关问题

genius-child-alice committed:
refactor(router): 移除获取待审核视频列表的路由

该路由功能已由其他接口实现，不再需要单独维护

genius-child-alice committed:
refactor(视频审核): 移除待审核视频相关功能代码

删除不再使用的待审核视频列表功能，包括控制器、服务层和DTO定义

YasaitoriOvO commented:
已拆分 故关闭
